src/main/java/VASSAL/tools/version/GitProperties.java

```diff
@@ -59,12 +62,19 @@ private void readGitProperties() {
   }
 
   public String getVersion() {
-    return String.format(
+    final String versionPrefix = String.format(
       "%s-%s-g%s",
       gitProperties.getProperty(KEY_GIT_CLOSEST_TAG_NAME, DEFAULT_CLOSEST_TAG_NAME),
       gitProperties.getProperty(KEY_GIT_CLOSEST_TAG_COMMIT_COUNT, DEFAULT_CLOSEST_TAG_COMMIT_COUNT),
       gitProperties.getProperty(KEY_GIT_COMMIT_ID, DEFAULT_COMMIT_ID).substring(0, GIT_COMMIT_ID_ABBREVIATED_LENGTH)
     );
+
+    final String versionSuffix = gitProperties.getProperty(KEY_GIT_BRANCH, DEFAULT_BRANCH);
+    if (!versionSuffix.equals(DEFAULT_BRANCH) && !versionSuffix.equals(SUPPRESS_BRANCH)) {
```

Comment:
I'm not following what this is intended to do w/r/t `DEFAULT_BRANCH`, as it's the empty string.

Reply:
If I don't check for DEFAULT_BRANCH here it would return a version string with a trailing "-"

Reply:
Oh, I see. I think what's confusing is that the variable is called `DEFAULT_BRANCH` when the value doesn't name a branch.